scripts/src/pullrequest/prepare_pr_comment.py

```diff
@@ -38,138 +43,245 @@ def get_look_at_job_output_comment():
 
 
+def overall_outcome(outcome):
+    return append_to("### Outcome:", f"**{outcome}**")
+
+
 def main():
     pr_content_result = sys.argv[1]
     run_verifier_result = sys.argv[2]
     verify_result = sys.argv[3]
     issue_number = open("./pr/NR").read().strip()
     vendor_label = open("./pr/vendor").read().strip()
     chart_name = open("./pr/chart").read().strip()
-    msg = get_comment_header(issue_number)
+
+    community_manual_review = os.environ.get("COMMUNITY_MANUAL_REVIEW", False)
     oc_install_result = os.environ.get("OC_INSTALL_RESULT", False)
 
+    msg = f"Thank you for submitting PR #{issue_number} for Helm Chart Certification!"
+
+    # Assemble the detail separately to control order in which it is added to
+    # the overall output.
+    detail_message = "### Detail"
+
+    outcome = "Failed"
+
     # Handle success explicitly
     if (
         pr_content_result == "success"
         and run_verifier_result == "success"
         and verify_result == "success"
         and oc_install_result == "success"
     ):
-        msg += prepare_success_comment()
+        outcome = "Passed"
+        detail_message = append_to(detail_message, get_success_coment())
         gitutils.add_output("pr_passed", "true")
     else:  # Handle various failure scenarios.
         if pr_content_result == "failure":
-            msg += prepare_pr_content_failure_comment()
+            detail_message = append_to(
+                detail_message, prepare_pr_content_failure_comment()
+            )
             gitutils.add_output("pr_passed", "false")
         elif run_verifier_result == "failure":
-            msg += prepare_run_verifier_failure_comment()
+            detail_message = append_to(
+                detail_message, prepare_run_verifier_failure_comment()
+            )
             gitutils.add_output("pr_passed", "false")
         elif verify_result == "failure":
-            community_manual_review = os.environ.get("COMMUNITY_MANUAL_REVIEW", False)
             if community_manual_review:
-                msg += prepare_community_comment()
+                outcome = "Pending Manual Review"
+                detail_message = append_to(detail_message, prepare_community_comment())
                 gitutils.add_output("pr_passed", "true")
             else:
-                msg += prepare_failure_comment()
+                detail_message = append_to(detail_message, prepare_failure_comment())
                 gitutils.add_output("pr_passed", "false")
         elif oc_install_result == "failure":
-            msg += prepare_oc_install_fail_comment()
+            detail_message = append_to(
+                detail_message, prepare_oc_install_fail_comment()
+            )
             gitutils.add_output("pr_passed", "false")
         else:
-            msg += prepare_generic_fail_comment()
+            detail_message = append_to(detail_message, prepare_generic_fail_comment())
             gitutils.add_output("pr_passed", "false")
```

Comment:
This is maybe out of scope, but looking at [this sandbox PR](https://github.com/openshift-helm-charts/sandbox/pull/71823), the message is not correct.

chart- verifier was skipped, outcome is Failed, we get a generic error message, but the PR is successfully merged.

Reply:
Looking at this before I push fixes out.

Reply:
Our logic is looking for an explicit failure, but in some cases we don't run the `install-oc` task and that shouldn't be considered a failure.

For now, I'm consolidating changes to this file but I think we may have opportunities for this to be fixed in a better way. As I shift to looking at the workflows (vs. this file), I may change this up. For now, I've allowed `"skipped"` and `False` as valid values for the OC installation, and `"skipped"` as a valid value for the run-verifier task because it may not be needed in all cases.

Reply:
Could you explain, and maybe add a comment in the code, why "False" is a valid result for OC install ?

Reply:
I should clarify: I don't like the use of False here. However, I believe its use is to be a third option other than passed or failed that isn't explicitly passed or failed. In other words, it doesn't trigger pass logic, and doesn't trigger fail logic, it just is. I would assume False is used to help with any logic cases where I need to say `if oc_install_result`. None may also be valid in that case. Either way, it was originally false, so I left it as is for the moment.

Reply:
Ok... A few things don't make complete sense to me, but those issues where already present prior to your PR.

I'll open a separate issue to discuss those.

Reply:
Opened #287 to handle separate issues I found regarding `oc-install`